crates/fluvio-test/src/tests/producer.rs

```diff
@@ -90,6 +89,11 @@ async fn producer_work(
     let mut throughput_histogram = Histogram::<u64>::new(2).unwrap();
     // Eventually we'll need to store globally to calculate wrt all producers in test
 
+    // We don't do anything with the record struct so we increase the stress on the system by just sending the same data over and over again.
+    // We can only do this because we don't compress the data
+    // Otherwise, test-harness spends much of the time allocating the record
+    let record = "A".repeat(test_case.option.record_size).as_bytes().to_vec();
```

Comment:
Workflow-wise, this is fine, but makes some small assumptions that affect the overall utility of the test. This test doesn't get used often, isn't in any automation, and the changes are net-positive. So I wouldn't do anything different at the moment.

This is more neutral feedback, I wouldn't hold up PR for this point because it is a bit out of the scope of your changes:

Enabling the test to be more configurable instead of removing functionality (even if this simple record form was made the default option) would have been my preference for record generation modes, instead of replacing one for the other. 

There are many producer options that have been introduced in the client that haven't been introduced into this test (like compression, and linger). And the ability to vary the shape of test data would further our ability to characterize performance.

Reply:
Agree 100%. Thanks for the feedback.